compiler/p/codegen/OMRTreeEvaluator.cpp

```diff
@@ -5149,37 +5267,10 @@ TR::Register *OMR::Power::TreeEvaluator::loadaddrEvaluator(TR::Node *node, TR::C
       else
          {
          int64_t  offset = mref->getOffset(*comp);
-         if (mref->hasDelayedOffset() || offset!=0 || comp->getOption(TR_AOT))
+         if (mref->hasDelayedOffset() || !mref->getBaseRegister() || offset!=0 || comp->getOption(TR_AOT))
             {
             resultReg = sym->isLocalObject() ? cg->allocateCollectedReferenceRegister() : cg->allocateRegister();
-            if (mref->hasDelayedOffset())
-               {
-               generateTrg1MemInstruction(cg, TR::InstOpCode::addi2, node, resultReg, mref);
-               }
-            else
-               {
-               if (offset>=LOWER_IMMED && offset<=UPPER_IMMED)
-                  {
-                  TR::Instruction *src2ForStatic;
-
-                  src2ForStatic = generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::addi2, node, resultReg, mref->getBaseRegister(), offset);
-                  if (mref->getStaticRelocation() != NULL)
```

Comment:
Yes, this code was all intentionally removed. This whole block of code existed only to try and use `Trg1Src1Imm` instructions instead of `Trg1Mem` instructions when possible, which is not necessary since both are functionally equivalent, and to handle some cases that are now handled in other locations (i.e. out-of-range displacements). Additionally, it was making assumptions about how the returned memory reference should look that are no longer true: it assumed that the displacement would never exceed `0x7fffffff`, that the base register would never be `NULL`, and didn't account for PC-relative references.

Since there's no need to even have special handling for these cases in the first place, I figured it would be better to remove all of this instead of trying to fix it, as fixing it would risk that it breaks in the future if we change `TR::MemoryReference` again.

Reply:
Yes, I specifically meant the relocation code. Is it take care of somewhere else?

Reply:
Ah, sorry, my bad. Yes, as long as the `TR::MemoryReference` is used to generate a `Trg1Mem` instruction, setting up the relo is taken care of during instruction expansion: https://github.com/eclipse/omr/blob/deecc7a4f0b3d0d05b47d5b7db35fce3d8f50d0a/compiler/p/codegen/OMRMemoryReference.cpp#L1276